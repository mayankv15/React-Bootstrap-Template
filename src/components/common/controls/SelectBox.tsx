import React, { useEffect, useState } from "react";
import { InputAdornment, Select as MuiSelect, MenuItem } from "@mui/material";
import Label from "./Label";
import COLORS from "../../../theme/colors.json";
import { useTranslation } from "react-i18next";
import { makeStyles } from "@mui/styles";

const useStyles = makeStyles(() => ({
  root: {
    display: "inline-flex",
    flexDirection: "column",
    width: "100%",
    textAlign: "center",
    "& .MuiOutlinedInput-input": {
      padding: "14px 30px 14px 5px",
    },
    "& .MuiInputBase-root ": {
      color: COLORS.TX_COLOR_5,
    },
    "& .Mui-focused": {
      "& .MuiSelect-select": {
        color: COLORS.APP_THEME_COLOR,
      },
    },
    "& .MuiSvgIcon-root": {
      color: COLORS.APP_THEME_COLOR,
    },
    "& .MuiOutlinedInput-notchedOutline": {
      color: COLORS.TX_COLOR_5,
    },
    "& .MuiOutlinedInput-root:hover": {
      color: COLORS.APP_THEME_COLOR,
    },
  },
}));

const SelectBox = (props) => {
  const {
    value,
    options,
    name,
    variant = "outlined",
    label,
    required,
    inputLabelStyles,
    handleChange,
    addIcon,
    classname,
  } = props;
  const classes = useStyles();
  const [selectValue, setSelectValue] = useState("default");
  const { t } = useTranslation();
  useEffect(() => {
    if (value) {
      setSelectValue(value);
    }
  }, [value, options]);

  // eslint-disable-next-line react-hooks/exhaustive-deps
  useEffect(() => {
    if (value && !selectValue) setSelectValue(value);
  });

  const handleSelectChange = (e) => {
    setSelectValue(e.target.value);
    if (handleChange) {
      handleChange(e);
    }
  };
  return (
    <div className={classes.root}>
      {label && (
        <Label
          title={label}
          required={required}
          inputLabelStyles={inputLabelStyles}
        />
      )}

      <MuiSelect
        className={addIcon ? "custom-select" : classname}
        value={value ? value : selectValue}
        variant={variant}
        name={name}
        defaultValue={value}
        onChange={handleSelectChange}
        inputProps={{
          startAdornment: (
            <InputAdornment position="start">sdsdsd</InputAdornment>
          ),
        }}
      >
        <MenuItem value="default" disabled hidden>
          --{t("select")}--
        </MenuItem>
        {options &&
          options.length > 0 &&
          options.length > 0 &&
          options.map((option) => (
            <MenuItem key={option.value} value={option.value}>
              {t(option.title)}
            </MenuItem>
          ))}
      </MuiSelect>
    </div>
  );
};

export default SelectBox;
