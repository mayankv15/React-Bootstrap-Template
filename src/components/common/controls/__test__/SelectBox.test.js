import React from "react";
import { render } from "@testing-library/react";
import SelectBox from "../SelectBox"

describe("Renders SelectBox", () => {
    it("should render SelectBox correctly", () => {
        const selectData = [
            {
                title: 'One',
                value: '1'
            },
            {
                title: 'Two',
                value: '2'
            },
            {
                title: 'Three',
                value: '3'
            },
        ]
        const component = render(<SelectBox
            label="Dropdown"
            options={selectData}
            variant="filled" />);
        expect(component).toMatchSnapshot();
    });
});
