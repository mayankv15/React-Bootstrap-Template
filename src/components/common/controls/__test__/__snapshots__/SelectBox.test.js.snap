// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders SelectBox should render SelectBox correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="makeStyles-root-1"
      >
        <p
          class="MuiTypography-root makeStyles-label-2 undefined MuiTypography-body1"
        >
          Dropdown
           
          
        </p>
        <div
          class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline"
        >
          <div
            aria-haspopup="listbox"
            class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
            role="button"
            tabindex="0"
          >
            --
            select
            --
          </div>
          <input
            aria-hidden="true"
            class="MuiSelect-nativeInput"
            startadornment="[object Object]"
            tabindex="-1"
            value="default"
          />
          <svg
            aria-hidden="true"
            class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M7 10l5 5 5-5z"
            />
          </svg>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="makeStyles-root-1"
    >
      <p
        class="MuiTypography-root makeStyles-label-2 undefined MuiTypography-body1"
      >
        Dropdown
         
        
      </p>
      <div
        class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline"
      >
        <div
          aria-haspopup="listbox"
          class="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-filled MuiInputBase-input MuiFilledInput-input"
          role="button"
          tabindex="0"
        >
          --
          select
          --
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          startadornment="[object Object]"
          tabindex="-1"
          value="default"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconFilled"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M7 10l5 5 5-5z"
          />
        </svg>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
